src/main/java/io/candydoc/ddd/extract_ddd_concepts/DDDConceptsExtractionService.java

```diff
@@ -64,72 +70,95 @@ private void trackAndApply(List<Event> occurredEvents) {
     occurredEvents.forEach(event -> event.accept(this));
   }
 
-  public void handle(ExtractValueObjects command) {
-    log.info("Extract value objects from {}", command.getPackageToScan());
-    trackAndApply(valueObjectExtractor.extract(command));
-  }
-
   public void handle(ExtractDDDConcepts command) {
     log.info("Extract ddd concepts from {}", command.getPackagesToScan());
-    trackAndApply(boundedContextExtractor.extract(command));
+    List<Event> boundedContextEvents = boundedContextExtractor.extract(command);
+    List<Event> sharedKernelEvents = sharedKernelExtractor.extract(command);
+    if (boundedContextEvents.isEmpty() && sharedKernelEvents.isEmpty()) {
+      throw new NoBoundedContextNorSharedKernelFound(command.getPackagesToScan());
+    }
+    trackAndApply(boundedContextEvents);
+    trackAndApply(sharedKernelEvents);
+  }
+
+  public void handle(ExtractAggregates command) {
+    log.info("Extract aggregates from {}", command.getPackageToScan());
+    trackAndApply(aggregatesExtractor.extract(command));
   }
 
+  public void handle(ExtractBoundedContexts command) {}
+
   public void handle(ExtractCoreConcepts command) {
     log.info("Extract core concepts from {}", command.getPackageToScan());
     trackAndApply(coreConceptExtractor.extract(command));
   }
 
+  public void handle(ExtractDomainCommands command) {
+    log.info("Extract domain commands from {}", command.getPackageToScan());
+    trackAndApply(domainCommandExtractor.extract(command));
+  }
+
   public void handle(ExtractDomainEvents command) {
     log.info("Extract domain events from {}", command.getPackageToScan());
     trackAndApply(domainEventExtractor.extract(command));
   }
 
-  public void handle(ExtractDomainCommands command) {
-    log.info("Extract domain commands from {}", command.getPackageToScan());
-    trackAndApply(domainCommandExtractor.extract(command));
+  public void handle(ExtractSharedKernels command) {}
```

Comment:
Why nothing happens ? I think there is something weird with the `handle(ExtractDDDConcepts command)`. It seems it has the responsabilities of the ExtractBoundedContexts and ExtractSharedKernels commands

Reply:
Or maybe the handler of each command should return an event relative to those cases. Something like ''NoBoundedContextFound'' and ''NoSharedKernelFound''.

If exactly those two events occurs, you can throw the expected error.

The other possibility is two say that if no event occurred, this is because neither BC nor SK have been found.

Reply:
So now we are checking in ExtractDDDConceptsUseCase if the list is empty or not.
List is empty is no bounded context or shared kernel has been found, if it is the case concepts will never be extracted.